lib/dotcom_web/controllers/stop_controller.ex

```diff
@@ -78,6 +86,52 @@ defmodule DotcomWeb.StopController do
     end
   end
 
+  defp banner_alerts(stop_id, route_ids, date_time) do
+    all_alerts_for_stop = @alerts_repo.by_stop_id(stop_id)
+
+    all_routewide_alerts =
+      @alerts_repo.by_route_ids(route_ids, date_time)
+      |> Enum.filter(&routewide?/1)
+
+    (all_alerts_for_stop ++ all_routewide_alerts)
+    |> Enum.filter(&banner_alert?(&1, date_time))
+  end
+
+  defp banner_alert?(alert, now) do
+    !global_banner_alert?(alert) &&
+      (banner_alert_active_or_future?(alert, now) || banner_alert_active_only?(alert, now))
+  end
+
+  defp banner_alert_active_or_future?(alert, now) do
+    banner_alert_active_or_future_effect?(alert) && active_in_next_n_days?(alert, 7, now)
+  end
+
+  defp banner_alert_active_or_future_effect?(alert) do
+    alert.effect in [
+      :dock_closure,
+      :dock_issue,
+      :service_change,
+      :shuttle,
+      :station_closure,
+      :station_issue,
+      :stop_closure,
+      :stop_moved,
+      :stop_shoveling,
+      :suspension
+    ]
+  end
```

Comment:
nit: defining _every_ single logic check into a separate named function is slightly hindering my ability to follow along, but I think this all checks out.

Reply:
Oh interesting - I found it easier to navigate this way! Probably low-urgency, but I'd be open to suggestions and/or happy to pair on making this easier to follow.